src/server/pegasus_server_impl.cpp

```diff
@@ -1187,21 +1187,23 @@ void pegasus_server_impl::on_get_scanner(get_scanner_rpc rpc)
 
         limiter->add_count();
 
-        auto state = append_key_value_for_scan(
-            resp.kvs,
+        auto state = validate_key_value_for_scan(
             it->key(),
             it->value(),
             request.hash_key_filter_type,
             request.hash_key_filter_pattern,
             request.sort_key_filter_type,
             request.sort_key_filter_pattern,
             epoch_now,
-            request.no_value,
-            request.__isset.validate_partition_hash ? request.validate_partition_hash : true,
-            return_expire_ts);
+            request.__isset.validate_partition_hash ? request.validate_partition_hash : true);
+
         switch (state) {
         case range_iteration_state::kNormal:
             count++;
+            if (!request.only_return_count) {
```

Comment:
Should judge wether request.__isset.only_return_count is set at first, client
maybe old version, it have to keep compatitable.